core/Scene.lua

```diff
@@ -53,10 +79,23 @@ function Scene:draw()
 end
 
 function Scene:update(dt)
+  if GS[self.level].state == 'GAMEOVER' then return end
+  if GS[self.level].state == 'LEVELDONE' then return end
   self.input_queue:update(dt)
+  GS[self.camera]:update(dt)
+
+  self.time_accumulator = self.time_accumulator and self.time_accumulator + dt or dt
+  if self.time_accumulator > 0.5 then
+    self.message_bus:publish('tick','tick',{})
+    self.time_accumulator = self.time_accumulator - 0.5
+  end
 end
 
 function Scene:input(channel, event)
+  local scene = self
+  event.begin = function(self)
+    GS[scene.gid].message_bus:publish(channel, event.name, event)
+  end
   self.input_queue:add(channel, event)
```

Comment:
The `input_queue` seems to complicate just sending messages to the message_bus. The time-scheduling is perhaps useful, but it brought no value to the LudumDare40 as it exists: it certainly didn't entice me to actually maintain a queue of player actions (the game went purely tick-based, and piped move commands immediately.)